src/lib/Libutil/misc_utils.c

```diff
@@ -1782,3 +1782,59 @@ perf_stat_stop(char *instance)
 
 	return (stat_summary);
 }
+
+/**
+ * @brief
+ *	creates an empty file in /tmp/ and saves timestamp of that file
+ *
+ * @param[in] - void
+ *
+ * @return - void
+ */
+void
+create_query_file(void)
+{
+	char filename[30];
+	char command[30];
+	snprintf(filename, 30, "/tmp/last_query_%d", getuid());
```

Comment:
Suggest you use $TMPDIR if it is defined in the environment. Also suggest you use pbs in the filename and make it a hidden file. Something like $TMPDIR/.pbs_query_%d